### ISSUE TITLE
function as parameter?

### ISSUE THREAD

**Issue by kitech** [vlang/v#498]
Seems nothing documented. A guessed syntax not work.

```
fn test(f fn()) {
}
```

**Comment by mingodad** [comment]
Good catch !

**Comment by medvednikov** [comment]
right now you can only do 

`fn test(f fn(int) int) {`

It has to have args or return values :) It's a bug. Will fix asap.

**Comment by medvednikov** [comment]
High order functions are still WIP, that's why there's no docs on them yet.

**Comment by Viza74** [comment]
Is there any plans when these will work?

I did some test, and only place function pointers/type is working in a function call as a parameter...
I would like to keep them in a variable or struct (for a finite state machine, or a simple entity system)

For testing I wrote a simple function:
```
fn fn_for_param(num int) int {
    println('$num funvar ')
    return num
}
```

Directly calling it like `fn_for_param(12)` obviously works and prints 12

Also work as a function parameter:
```
fn testfun(f fn(int) int) {
	f(13)
}

...and later...
testfun(fn_for_param)
```
prints 13 as expected.
(BTW testfun happily accepts a function parameter even if it is not matching the signature for example
```
fn fn_for_param(num int) {
    println('$num funvar ')
}
```
is still works and prints 13 , it probably should give an error?)

But these won't work:
```
mut f := fn_for_param
f(14)
```
or
```
mut f := &fn_for_param
f(14)
```

and also not working in a struct:
```
	state := gfrm.FSMState{
		enter: fn_for_param
		update: fn_for_param
		exit: fn_for_param
	}

	state.enter(90)
	state.update(91)
	state.exit(92)
```
These print nothing, and also there are no any other indications that these are even there - no warnings or errors.
